animata/list/list-stack.tsx

```diff
@@ -0,0 +1,167 @@
+        </div>
+      </div>
+    </motion.div>
+  );
+};
+
+export const ListStack = ({ cards, offset, scaleFactor }: ListStackProps) => {
+  const CARD_OFFSET = offset || 9;
+  const SCALE_FACTOR = scaleFactor || 0.03;
+  const [showAll, setShowAll] = useState(false);
+
+  const toggleShow = () => setShowAll(!showAll);
+  const middleIndex = Math.floor(cards.length / 2);
+
+  const revealAnimation = {
+    hidden: { opacity: 0, scale: 1 },
+    visible: {
+      opacity: 1,
+      transition: {
+        duration: 0.5,
+        ease: [0, 0.71, 0.2, 1.01],
+        scale: {
+          type: "spring",
+          damping: 5,
+          stiffness: 100,
+          restDelta: 0.001,
+        },
+      },
+    },
+  };
+
+  const stackAnimation = {
+    hidden: { opacity: 0, scale: 1 },
+    visible: (i: number) => ({
+      opacity: 1,
+      y: (i - middleIndex) * CARD_OFFSET,
+      scale: 1 - i * SCALE_FACTOR,
+      zIndex: cards.length - i,
+    }),
+  };
+
+  return (
+    <div className="mx-auto flex flex-col items-center justify-center">
+      <div className="relative flex w-full justify-center">
+        <div className="relative min-h-24">
+          <div className="flex flex-col items-center">
+            <AnimatePresence>
+              {showAll
+                ? cards.map((card, index) => (
+                    <motion.div
+                      key={card.id}
+                      custom={index}
+                      variants={revealAnimation}
+                      initial="hidden"
+                      animate="visible"
+                      exit="hidden"
+                      className="mb-3"
+                    >
+                      <CardStack
+                        id={card.id}
+                        icon={card.icon}
+                        title={card.title}
+                        location={card.location}
+                        date={card.date}
+                      />
+                    </motion.div>
+                  ))
+                : cards.map((card, index) => (
+                    <motion.div
+                      key={card.id}
+                      custom={index}
+                      variants={stackAnimation}
+                      initial="hidden"
+                      animate="visible"
+                      exit="hidden"
+                      style={{
+                        position: "absolute",
+                        transformOrigin: "top center",
+                        zIndex: cards.length - index,
+                      }}
+                    >
+                      <CardStack
+                        id={card.id}
+                        icon={card.icon}
+                        title={card.title}
+                        location={card.location}
+                        date={card.date}
+                      />
+                    </motion.div>
+                  ))}
+            </AnimatePresence>
+          </div>
```

Comment:
You don't need the conditional rendering here. Just conditionally change the CSS style (transform).